server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1401,7 +1555,7 @@ private void executeOneStaleIndexDelete(
                     logger.warn(new ParameterizedMessage("[{}] Exception during single stale index delete", metadata.name()), e);
```

Comment:
Here it is ignoring any failure during deletion of index blobs as those are stale and can be deleted in next try. However, for the lock file case, if there is any failure in the release of lock files we will need to fail the operation. Otherwise the lock file will still remain however the index level folder will get deleted, thus leaking all the lock files.